apps/src/gamelab/shapes.js

```diff
@@ -182,13 +182,20 @@ export function throwIfSerializedAnimationListIsInvalid(serializedAnimationList)
 
   // Check for duplicates in the orderedKeys array
   let knownKeys = {};
-  orderedKeys.forEach(key => {
+  let dupIndices = [];
+  orderedKeys.forEach((key, index) => {
     if (knownKeys.hasOwnProperty(key)) {
-      throw new Error(`Key "${key}" appears more than once in orderedKeys`);
+      // Store indices from highest to lowest to make it easier to remove dupes
+      dupIndices.splice(0, 0, index);
     }
     knownKeys[key] = true;
   });
 
+  // Remove any duplicated indices
+  dupIndices.forEach(index => {
+    orderedKeys.splice(index, 1);
+  });
```

Comment:
your code here looks good, but for utility method operations like this, I'd always rather use an existing library, in this case `_.uniq`: https://lodash.com/docs/4.17.5#uniq